@here/harp-map-controls/lib/MapControls.ts

```diff
@@ -1113,19 +1147,25 @@ export class MapControls extends THREE.EventDispatcher {
         const toWorld = MapViewUtils.rayCastWorldCoordinates(this.mapView, to.x, to.y);
         const fromWorld = MapViewUtils.rayCastWorldCoordinates(this.mapView, from.x, from.y);
 
-        if (!toWorld || !fromWorld) {
+        if (toWorld === null || fromWorld === null) {
             return;
         }
 
-        this.m_panDistanceFrameDelta = fromWorld.sub(toWorld);
+        if (this.mapView.projection.type === geoUtils.ProjectionType.Planar) {
+            // Cancel zoom inertia if a panning is triggered, so that the mouse location is kept.
+            this.stopZoom();
 
-        // Cancel zoom inertia if a panning is triggered, so that the mouse location is kept.
-        this.stopZoom();
+            // Assign the new animation start time.
+            this.m_panAnimationStartTime = performance.now();
 
-        // Assign the new animation start time.
-        this.m_panAnimationStartTime = performance.now();
+            this.m_panDistanceFrameDelta = fromWorld.sub(toWorld);
 
-        this.pan();
+            this.pan();
+        } else {
+            quaternion.setFromUnitVectors(fromWorld.normalize(), toWorld.normalize()).inverse();
```

Comment:
This should go to MapControls.pan. Will there be inertia for globe panning(spinning) ?

Reply:
Yes. Problem is that the pan function receives a vector in world units, which doesn't work for the globe, at best we want a delta in geolocations for a common approach, which requires some change in the structure of MapControls. I want to separate these questions into tickets and tackle them separately, so that the controls are an abstraction of the movement that works for all projections and where inertia handle both the same way too, yes.